x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -649,11 +649,16 @@ function _enforceFrozenVars(
   newVars: Record<string, PackagePolicyConfigRecordEntry>
 ) {
   const resultVars: Record<string, PackagePolicyConfigRecordEntry> = {};
+  for (const [key, val] of Object.entries(newVars)) {
+    if (oldVars[key]?.frozen) {
+      resultVars[key] = oldVars[key];
+    } else {
+      resultVars[key] = val;
+    }
+  }
   for (const [key, val] of Object.entries(oldVars)) {
```

Comment:
I believe so, because if you had in `oldVars` fields that were frozen and then you send `newVars` without including them I believe we still want to keep those frozen fields, what do you think?